File: src/R/mediawiki_parser.R
#' Convert wikitext to HTML using MediaWiki's parser
#'
#' @param wikitext Character string containing wikitext markup
#' @param format Output format ('html' or 'json')
#' @return If format is 'html', returns HTML string. If format is 'json',
#'         returns a list containing parsed text and metadata
#' @export
convert_wikitext <- function(wikitext, format = c("html", "json")) {
    format <- match.arg(format)
    
    # Escape the wikitext for shell command
    escaped_wikitext <- gsub('"', '\\"', wikitext)
    
    # Build the command
    script_path <- file.path(dirname(sys.frame(1)$ofile), "..", "mediawiki_parser.php")
    
    # Execute the command
    result <- tryCatch({
        system2("php", args = c(
            script_path,
            "--wikitext", escaped_wikitext,
            "--format", format
        ), stdout = TRUE, stderr = TRUE)
        
        if (format == "json") {
            jsonlite::fromJSON(paste(result, collapse = "\n"))
        } else {
            paste(result, collapse = "\n")
        }
    }, error = function(e) {
        stop("Error executing parser: ", e$message)
    })
    
    return(result)
}

#' Convert wikitext to HTML with full MediaWiki parsing
#'
#' @param wikitext Character string containing wikitext markup
#' @return HTML string
#' @export
convert_wikitext_to_html <- function(wikitext) {
    convert_wikitext(wikitext, format = "html")
}

#' Convert wikitext to HTML and get metadata
#'
#' @param wikitext Character string containing wikitext markup
#' @return List containing parsed HTML and metadata (links, images, etc.)
#' @export
convert_wikitext_with_metadata <- function(wikitext) {
    convert_wikitext(wikitext, format = "json")
}

File: src/mediawiki_parser.php
<?php
/**
 * Standalone MediaWiki Parser
 * 
 * This script provides a standalone interface to MediaWiki's parser
 * Usage: php mediawiki_parser.php --wikitext="'''bold''' text" [--format=html|json]
 */

// Set up autoloading for MediaWiki classes
require_once __DIR__ . '/../vendor/autoload.php';
require_once __DIR__ . '/../lib/includes/AutoLoader.php';

use MediaWiki\MediaWikiServices;
use MediaWiki\Parser\Parser;
use MediaWiki\Parser\ParserOptions;
use MediaWiki\Parser\ParserOutput;
use MediaWiki\Title\Title;
use MediaWiki\User\User;

class StandaloneParser {
    private $services;
    private $parser;
    private $parserOptions;
    
    public function __construct() {
        // Initialize MediaWiki's service container
        $this->services = MediaWikiServices::getInstance();
        
        // Initialize parser
        $this->parser = $this->services->getParser();
        
        // Create default user and title context
        $user = new User();
        $title = Title::newFromText('Standalone_Parser_Page');
        
        // Initialize parser options
        $this->parserOptions = ParserOptions::newFromUser($user);
        $this->parserOptions->setTidy(true);
        $this->parserOptions->setRemoveComments(true);
    }
    
    /**
     * Parse wikitext to HTML
     */
    public function parse($wikitext) {
        // Create a title object for context
        $title = Title::newFromText('Standalone_Parser_Page');
        
        // Parse the text
        $output = $this->parser->parse(
            $wikitext,
            $title,
            $this->parserOptions
        );
        
        return $output->getText();
    }
    
    /**
     * Parse wikitext and return full parsing results
     */
    public function parseWithMetadata($wikitext) {
        $title = Title::newFromText('Standalone_Parser_Page');
        $output = $this->parser->parse(
            $wikitext,
            $title,
            $this->parserOptions
        );
        
        return [
            'text' => $output->getText(),
            'links' => $output->getLinks(),
            'images' => $output->getImages(),
            'categories' => $output->getCategories(),
            'templates' => $output->getTemplates(),
            'sections' => $output->getSections(),
            'properties' => $output->getProperties(),
        ];
    }
}

// CLI interface
if (PHP_SAPI === 'cli') {
    // Parse command line arguments
    $options = getopt('', ['wikitext:', 'format:']);
    
    if (!isset($options['wikitext'])) {
        die("Usage: php mediawiki_parser.php --wikitext=\"'''bold''' text\" [--format=html|json]\n");
    }
    
    $format = $options['format'] ?? 'html';
    $wikitext = $options['wikitext'];
    
    try {
        $parser = new StandaloneParser();
        
        if ($format === 'json') {
            $result = $parser->parseWithMetadata($wikitext);
            echo json_encode($result, JSON_PRETTY_PRINT);
        } else {
            $result = $parser->parse($wikitext);
            echo $result;
        }
        echo "\n";
    } catch (Exception $e) {
        die("Error: " . $e->getMessage() . "\n");
    }
}